### ISSUE TITLE
Should use less ! and do more inline error checking and form redisplay

### ISSUE THREAD

**Issue by cg2v** [autolab/Autolab#885]
There are many situations where autolab uses the ! form of an activerecord function, which will trigger donkey kong if a validation or other error occurs, instead of catching the validation error and redisplaying whatever form triggered it. Recent high-profile examples of this include all the error emails we've been getting about "You cannot submit because it is past the deadline" and some earlier errors when the the end time of an assessment was not updated when the start time changed.

**Comment by zyx-billy** [comment]
Agree. Overall, the entire site needs more error handlers instead of simply crashing on the user. It's really frustrating when editing an assessment, one error causes all changes to be discarded. However, since there's many such problems, we'll need to focus on the more important and frequently occurring ones. But at least when writing new code, we should keep this in mind.
